homeassistant/components/electric_kiwi/__init__.py

```diff
@@ -70,3 +74,35 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    # convert title and unique_id to string
+    if config_entry.version == 1 and config_entry.minor_version == 0:
+        if config_entry.unique_id is not None and config_entry.unique_id.startswith(
+            DOMAIN
+        ):
+            implementation = (
+                await config_entry_oauth2_flow.async_get_config_entry_implementation(
+                    hass, config_entry
+                )
+            )
+
+            session = config_entry_oauth2_flow.OAuth2Session(
+                hass, config_entry, implementation
+            )
+
+            ek_api = ElectricKiwiApi(
+                api.ConfigEntryElectricKiwiAuth(
+                    aiohttp_client.async_get_clientsession(hass), session
+                )
+            )
+
+            ek_session = await ek_api.get_active_session()
+            unique_id = "_".join(str(num) for num in ek_session.customer_numbers)
+            hass.config_entries.async_update_entry(
+                config_entry, unique_id=unique_id, title=unique_id, version=2
```

Comment:
We should avoid updating titles as they are considered user land

Reply:
Well the current title is "Home Assistant Cloud" so I think in this case it's a valid change to their customer number

Reply:
But users could have changed them

Reply:
Oh good point. Will remove that.